[docs] add compilation bits to the bitsandbytes docs. (#11693)
huggingface/diffusers: docs/source/en/quantization/bitsandbytes.md

transformer_4bit.dequantize()
```

> ## torch.compile compatibility

Reviewer: ```suggestion
## torch.compile
```

<hfoptions id="bnb">
<hfoption id="8-bit">

> You can speed up inference of `bitsandbytes`-quantized models with `torch.compile()`. Make sure you have
> the latest `bitsandbytes` installed - `pip install -U bitsandbytes`. We also recommend installting PyTorch
> nightlies. Follow instructions [here](https://pytorch.org/).

Reviewer: I would place this outside the `<hfoptions>` blocks and under `## torch.compile` since it applies to both 8 and 4-bit.

```suggestion
Speed up inference with `torch.compile`. Make sure you have
the latest `bitsandbytes` installed and we also recommend installing [PyTorch nightly](https://pytorch.org/get-started/locally/).

```py
pip install -U bitsandbytes
```
```

```py
torch._dynamo.config.capture_dynamic_output_shape_ops = True

quant_config = DiffusersBitsAndBytesConfig(load_in_8bit=True)
transformer_4bit = AutoModel.from_pretrained(
    "black-forest-labs/FLUX.1-dev",
    subfolder="transformer",
    quantization_config=quant_config,
    torch_dtype=torch.float16,
)
transformer_4bit.compile(fullgraph=True)
```

</hfoption>
<hfoption id="4-bit">

```py
quant_config = DiffusersBitsAndBytesConfig(load_in_4bit=True)
transformer_4bit = AutoModel.from_pretrained(
    "black-forest-labs/FLUX.1-dev",
    subfolder="transformer",
    quantization_config=quant_config,
    torch_dtype=torch.float16,
)
transformer_4bit.compile(fullgraph=True)
```
</hfoption>
</hfoptions>

> We benchmarked performance of 4bit Flux with and without compilation and noticed a massive improvement on RTX 4090:
> 
> * Without compilation: 32.570 seconds
> * With compilation: 25.809 seconds

Reviewer: ```suggestion
On a RTX 4090 with compilation, 4-bit Flux generation completed in 25.809 seconds versus 32.570 seconds without.
```

> Check out the script used for benchmarking [here](https://gist.github.com/sayakpaul/0db9d8eeeb3d2a0e5ed7cf0d9ca19b7d).

Reviewer: ```suggestion
Check out the [benchmarking script](https://gist.github.com/sayakpaul/0db9d8eeeb3d2a0e5ed7cf0d9ca19b7d) for more details.
```
